src/tests/ftest/rebuild/container_create.py

```diff
@@ -0,0 +1,133 @@
+                                                   record_qty, akey_size,
+                                                   dkey_size, data_size, rank,
+                                                   obj_class, self.d_log)
+        except DaosTestError as excep:
+            self.fail("Failed to write initial objects to pool: "
+                      "{0}".format(str(excep)))
+
+        # Debug
+        self.log.info("Checking pool status prior to rebuild")
+        get_pool_status(self.pool, self.log)
+
+        # Kill the server
+        self.log.info(
+            "Killing DAOS server %s (rank %s)", self.server_group, rank)
+        kill_server(self.server_group, self.context, rank, self.pool,
+                    self.d_log)
+
+        # Wait for recovery to start
+        wait_for_rebuild(self.pool, self.log, True, 1)
+
+        # make 2nd container during rebuild
+        container2 = get_container(self.context, self.pool, self.d_log)
+
+        # write a single object to it
+        try:
+            rebuild_write = write_single_objects(container2, 1, 1, 5, 5, 10,
+                                                 rebuild_write_rank, obj_class)
+        except DaosTestError as excep:
+            self.fail("Failed to write object to second container: "
+                      "{0}".format(str(excep)))
+
+        # wait for rebuild to complete
+        wait_for_rebuild(self.pool, self.log, False, 1)
+
+        # verify data rebuilt correctly
+        for obj_dict in written_objects:
+            for rec_dict in obj_dict["record"]:
+                read_data = read_single_objects(self.container, data_size,
+                                                rec_dict["dkey"],
+                                                rec_dict["akey"],
+                                                obj_dict["obj"],
+                                                obj_dict["txn"])
+                if read_data != rec_dict["data"]:
+                    self.fail("Data post-rebuild does not match pre-rebuild.\n"
+                              "Read data: {0}\n"
+                              "Expected data: {1}".format(read_data,
+                                                          rec_dict["data"]))
+
+        # verify the data written to 2nd container during rebuild is readable
+        rebuild_read = read_single_objects(
+            container2, 10,
+            rebuild_write[0]["record"][0]["dkey"],
+            rebuild_write[0]["record"][0]["akey"],
+            rebuild_write[0]["obj"],
+            rebuild_write[0]["txn"])
+
+        if rebuild_read != rebuild_write[0]["record"][0]["data"]:
+            self.fail("Data written to second container does not match "
+                      "expected data.\n"
+                      "Read data: {0}\n"
+                      "Expected data: {1}".format(rebuild_read,
+                                                  rebuild_write[0]["data"]))
```

Comment:
I'm about to add this to my patch.  We should verify the number of rebuilt records after rebuild is complete:

        # Confirm the number of rebuilt objects reported by the pool query
        expected_pool_info = {
            "rs_toberb_obj_nr": object_qty,
            "rs_obj_nr": object_qty,
            "rs_rec_nr": object_qty * record_qty * data_size,
            "rs_errno": 0
        }
        self.log.info("Verifying the number of rebuilt objects and status")
        pool_info = get_pool_status(self.pool, self.log)
        for key, expected in expected_pool_info.items():
            detected = getattr(pool_info.pi_rebuild_st, key)
            self.assertEqual(
                detected, expected,
                "Unexpected {} value: expected={}, detected={}".format(
                    key, expected, detected))

Reply:
I've actually changed the implementation to add this as a function in general_utils.py in the latest commit in #467

Reply:
Added same function.